### PR TITLE
groq[major]: ChatGroq (groq integration package)

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

**Comment by gradenr** [comment]
For the python integration I was asked to add a provider doc page.  I assume there's something similar for JS.

Here's an example from my PR:
vercel deploy: https://langchain-lu1zuf8nu-langchain.vercel.app/docs/integrations/providers/groq
mdx file: https://github.com/langchain-ai/langchain/pull/17856/files#diff-7f412d0a3b880dd49da4652a288a5fc50fe7b28dba973ee4626c18ab29a3d285

File: libs/langchain-groq/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+const model = new ChatGroq({
+  apiKey: process.env.GROQ_API_KEY, // Default value.
+});
+
+const message = new HumanMessage("What color is the sky?");
+
+const res = await chat.invoke([message]);
+```
+
+## Development
```
Comment (gradenr) [inline libs/langchain-groq/README.md:34]: Nice development section.  I'm going to add this to the python readme.

File: libs/langchain-groq/src/chat_models.ts [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    messages: BaseMessage[],
+    options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun
+  ): Promise<ChatResult> {
+    const tokenUsage: TokenUsage = {};
+    const params = this.invocationParams(options);
+    const messagesMapped = convertMessagesToGroqParams(messages);
+
+    if (params.stream) {
+      console.log("streaming via _generate");
```
Comment (jacoblee93) [inline libs/langchain-groq/src/chat_models.ts:272]: Remove log

**Comment by jacoblee93** [comment]
> For the python integration I was asked to add a provider doc page. I assume there's something similar for JS.
> 
> Here's an example from my PR: vercel deploy: https://langchain-lu1zuf8nu-langchain.vercel.app/docs/integrations/providers/groq mdx file: https://github.com/langchain-ai/langchain/pull/17856/files#diff-7f412d0a3b880dd49da4652a288a5fc50fe7b28dba973ee4626c18ab29a3d285

At the moment, just the chat model page for JS! You're all good.

**Review by gradenr — APPROVED** [review]
Thanks for getting this done for us.

File: libs/langchain-groq/src/chat_models.ts [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+      throw new Error(
+        `Groq API key not found. Please set the GROQ_API_KEY environment variable or provide the key into "apiKey"`
+      );
+    }
+
+    this.client = new Groq({
+      apiKey,
+    });
+    this.temperature = fields?.temperature ?? this.temperature;
+    this.modelName = fields?.modelName ?? this.modelName;
```
Comment (HenryHengZJ) [inline libs/langchain-groq/src/chat_models.ts:194]: can we pass the `stream` here?
`this.stream = fields?.stream ?? this.stream;`

Comment (bracesproul) [reply libs/langchain-groq/src/chat_models.ts:194]: Hey, the recommended way of streaming is via the `.stream` method. Although this is a good catch, we should be passing the `streaming` class property here. Will add it today

Comment (HenryHengZJ) [reply libs/langchain-groq/src/chat_models.ts:194]: are we deprecating the `streaming` property as well (for other chat models)?

Comment (jacoblee93) [reply libs/langchain-groq/src/chat_models.ts:194]: @bracesproul we should add this - there are niche but legit use cases for invoke with aggregating token information.

Comment (bracesproul) [reply libs/langchain-groq/src/chat_models.ts:194]: Will be in soon! https://github.com/langchain-ai/langchainjs/pull/4573/files

**Pull request merged**